prime-router/src/main/kotlin/azure/TokenFunction.kt

```diff
@@ -91,12 +89,8 @@ class TokenFunction(val metadata: Metadata = Metadata.getInstance()) : Logging {
         } else {
             actionHistory.trackActionResult("Token request denied.")
             actionHistory.setActionType(TaskAction.token_error)
-            if (senderKeyFinder.errorMsg != null) {
-                logger.error("${senderKeyFinder.errorMsg}")
-                actionHistory.trackActionResult("${senderKeyFinder.errorMsg}")
-            }
 
-            HttpUtilities.unauthorizedResponse(request)
+            HttpUtilities.unauthorizedResponse(request, "invalid_client")
```

Comment:
The [spec](http://hl7.org/fhir/uv/bulkdata/authorization/index.html#protocol-details:~:text=the%20server%20SHALL%20respond%20with%20an%20invalid_client%20error) says that if auth fails this should be the response.